File: medplum-java/medplum-model/src/main/java/com/fhir/MessageHeader.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMessageHeader.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MessageHeader {
  Optional<Reference> author();

  MessageHeader_Source source();

  Optional<List<ResourceList>> contained();

  Optional<List<Extension>> modifierExtension();

  Optional<canonical> definition();

  Optional<Narrative> text();

  Optional<String> eventUri();

  Optional<List<MessageHeader_Destination>> destination();

  Optional<Reference> sender();

  Optional<code> language();

  Optional<Coding> eventCoding();

  Optional<uri> implicitRules();

  String resourceType();

  Optional<List<Extension>> extension();

  Optional<Reference> enterer();

  Optional<CodeableConcept> reason();

  Optional<MessageHeader_Response> response();

  Optional<List<Reference>> focus();

  Optional<id> id();

  Optional<Meta> meta();

  Optional<Reference> responsible();

  static ImmutableMessageHeader.SourceBuildStage builder() {
    return ImmutableMessageHeader.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MessageHeader.ts
import { canonical } from "./canonical";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Coding } from "./Coding";
import { Extension } from "./Extension";
import { id } from "./id";
import { MessageHeader_Destination } from "./MessageHeader_Destination";
import { MessageHeader_Response } from "./MessageHeader_Response";
import { MessageHeader_Source } from "./MessageHeader_Source";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface MessageHeader {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    eventCoding: Coding | null | undefined;
    eventUri: string | null | undefined;
    destination: MessageHeader_Destination[] | null | undefined;
    sender: Reference | null | undefined;
    enterer: Reference | null | undefined;
    author: Reference | null | undefined;
    source: MessageHeader_Source;
    responsible: Reference | null | undefined;
    reason: CodeableConcept | null | undefined;
    response: MessageHeader_Response | null | undefined;
    focus: Reference[] | null | undefined;
    definition: canonical | null | undefined;
}
